### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by morris-jason**
Addresses most concerns of #1090.
Improves logging and speed of the ova builder. Refactors build scripts into area-specific bash scripts with optional caching of the base vmdk.

### CODE REVIEW

**Comment by morris-jason** [comment]
Green builds: https://ci-vic.vmware.com/vmware/vic-product/4252 https://ci-vic.vmware.com/vmware/vic-product/4253

File: installer/build/bootable/build-main.sh [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    esac
+done
+shift $((OPTIND-1))
+# check there were no extra args and the required ones are set
+if [ -n "$*" -o -z "${RESOURCE}" -o -z "${MANIFEST}" ]; then
+    usage
+fi
+
+exec 3>&1 1>>"${RESOURCE}/installer-build.log" 2>&1
+log1 "Staring appliance build."
```
Comment (andrewtchin) [inline installer/build/bootable/build-main.sh]: starting

File: installer/docs/BUILD.md [hunk 2]
```diff
@@ -61,6 +62,23 @@ specified by their respective URLs, Admiral tag `vic_v1.1.1`, and VIC Machine Se
...
+The `bootable` folder contains all the files needed to make a bootable ova. These include `build-main.sh`, which organizes the calls for `build-disks.sh`, `build-base.sh`, and `build-app.sh`. 
+
+These three scripts are self-explanatory:
+ - `build-disks.sh`: Provisions local disk space for the boot and data drives. Installs grub2 to the boot drive.
+ - `build-base.sh`: Installs all repo components, like a linux kernel and coreutils, to the base disks. Can be cached as a gzipped tar.
+ - `build-app.sh`: Performs any necessary configuration of the ova by running all script provisioners in a chroot.
+
+The `bootable` folder also contains ovf template and tdnf repos for building the ova.
+
+There are many useful arguments for `build-main.sh`, but most notable is the `-b` argument for cachine the base layer for faster builds. This option can be passed throught the first `build.sh` script, like `./build/build.sh ova-dev -b bin/.vic-appliance-base.tar.gz`.
```
Comment (andrewtchin) [inline installer/docs/BUILD.md]: caching

File: installer/build/bootable/build-main.sh (hunk 1 above)
Comment (morris-jason) [reply installer/build/bootable/build-main.sh]: Done.

File: installer/docs/BUILD.md (hunk 2 above)
Comment (morris-jason) [reply installer/docs/BUILD.md]: Done.

**Pull request merged**